### PR TITLE
fix(sdk): exception handling

### PR DESCRIPTION
**Pull request by bojiang**
## Description

Avoid unexcepted key errors when an exception happened in the runner.

## Checklist:

- [ ] My code follows the bentoml code style, both `make format` and
  `make lint` script have passed
  ([instructions](https://github.com/bentoml/BentoML/blob/master/DEVELOPMENT.md#style-check-and-auto-formatting-your-code)).
- [ ] My change reduces project test coverage and requires unit tests to be added
- [ ] I have added unit tests covering my code change
- [ ] My change requires a change to the documentation
- [ ] I have updated the documentation accordingly

### CODE REVIEW

File: bentoml/_internal/io_descriptors/json.py [hunk 1]
```diff
@@ -154,19 +154,23 @@ def openapi_responses_schema(self) -> t.Dict[str, t.Any]:
...
         if self._pydantic_model is not None and self._validate_json:
             try:
                 import pydantic
             except ImportError:
                 raise MissingDependencyException(
                     "`pydantic` must be installed to use `pydantic_model`"
-                )
+                ) from None
+            try:
+                return self._pydantic_model.parse_obj(json_str)
```
Comment (parano) [inline bentoml/_internal/io_descriptors/json.py:166]:
```suggestion
                return self._pydantic_model.parse_raw(json_str)
```

File: bentoml/_internal/io_descriptors/json.py [hunk 2]
```diff
@@ -154,19 +154,23 @@ def openapi_responses_schema(self) -> t.Dict[str, t.Any]:
         return {MIME_TYPE_JSON: {"schema": self.openapi_schema_type()}}
 
     async def from_http_request(self, request: Request) -> JSONType:
-        json_obj = await request.json()
+        json_str = await request.body()
```
Comment (parano) [inline bentoml/_internal/io_descriptors/json.py:157]: what's the reason for this change?

File: tests/e2e/bento_server_general_features/service.py [hunk 3]
```diff
@@ -53,6 +53,11 @@ class _Schema(pydantic.BaseModel):
     function_name="echo_multi_ndarray",
     name="echo_multi_ndarray_pred_runner",
 )
+raise_user_error_runner = bentoml.sklearn.load_runner(
```
Comment (parano) [inline tests/e2e/bento_server_general_features/service.py:56]: should we migrate testing to use picklable_model?

File: bentoml/_internal/io_descriptors/json.py (hunk 2 above)
Comment (bojiang) [reply bentoml/_internal/io_descriptors/json.py:157]: `request.json` may raise exceptions
